ihe-interop/src/main/java/org/motechproject/ihe/interop/service/impl/IHETemplateServiceImpl.java

```diff
@@ -30,4 +32,21 @@ public void sendTemplateToRecipientUrl(String url, String template) throws IOExc
         int responseCode = client.executeMethod(post);
         LOGGER.info("Response code: {}", responseCode);
     }
+
+    @Override
+    public void sendTemplateToRecipientUrlWithBA(HL7Recipient recipient, String template) throws IOException {
+
+        PostMethod post = new PostMethod(recipient.getRecipientUrl());
+        HttpClient client = new HttpClient();
+        UsernamePasswordCredentials credentials = new UsernamePasswordCredentials(recipient.getRecipientUsername(),
+                recipient.getRecipientPassword());
+
+        post.setRequestEntity(new StringRequestEntity(template, "application/xml", "utf-8"));
+        client.getParams().setAuthenticationPreemptive(true);
+        client.getState().setCredentials(AuthScope.ANY, credentials);
+
+        LOGGER.info("Sending template with Basic Authentication to URL: {}", recipient.getRecipientUrl());
+        int responseCode = client.executeMethod(post);
+        LOGGER.info("Response code: {}", responseCode);
```

Comment:
It seems like a lot of this is repeated from the method above. Can we have a private method(s) for the common code?